### PR TITLE
fix: Use MarshalText when serializing timestamps when applicable

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Proper fix for https://github.com/cloudquery/plugin-sdk/pull/373

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: schema/timestamptz.go [hunk 1]
```diff
@@ -90,6 +94,12 @@ func (dst *Timestamptz) Set(src interface{}) error {
 		if originalSrc, ok := underlyingTimeType(src); ok {
 			return dst.Set(originalSrc)
 		}
+		if value, ok := value.(TextMarshaler); ok {
+			s, err := value.MarshalText()
+			if err == nil {
```
Comment (candiduslynx) [inline schema/timestamptz.go:99]: What happens if `err` is not nil? Do we lose this info?

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: schema/timestamptz.go [hunk 2]
```diff
@@ -24,6 +24,10 @@ type TimestamptzTransformer interface {
 	TransformTimestamptz(*Timestamptz) interface{}
 }
 
+type TextMarshaler interface {
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply schema/timestamptz.go:27]: Ah nice I'll use that

File: schema/timestamptz.go (hunk 1 above)
Comment (erezrokah) [reply schema/timestamptz.go:99]: Yeah, we can't really print a warning and using `String()` might just work

File: schema/timestamptz_test.go [hunk 3]
```diff
@@ -21,19 +30,16 @@ func TestTimestamptzSet(t *testing.T) {
 		{source: _time(time.Date(1970, 1, 1, 0, 0, 0, 0, time.Local)), result: Timestamptz{Time: time.Date(1970, 1, 1, 0, 0, 0, 0, time.Local), Status: Present}},
 		{source: Infinity, result: Timestamptz{InfinityModifier: Infinity, Status: Present}},
 		{source: NegativeInfinity, result: Timestamptz{InfinityModifier: NegativeInfinity, Status: Present}},
-		{source: "2105-07-23 22:23:37.133334968 +0300 UTC m=+2610529275.484148261", result: Timestamptz{Time: time.Date(2105, 7, 23, 22, 23, 37, 133334968, time.UTC), Status: Present}},
-		// {source: "2020-04-05 06:07:08Z", result: Timestamptz{Time: time.Date(2020, 4, 5, 6, 7, 8, 0, time.UTC), Status: Present}},
+		{source: string(timeRFC3339NanoBytes), result: Timestamptz{Time: time.Date(2105, 7, 23, 22, 23, 37, 750076110, time.UTC), Status: Present}},
```
Comment (erezrokah) [inline schema/timestamptz_test.go:33]: Re-added the reverted tests and added a new one for a wrapped `time.Time`

**Review by candiduslynx — APPROVED** [review]

File: schema/timestamptz_test.go [hunk 4]
```diff
@@ -21,19 +30,16 @@ func TestTimestamptzSet(t *testing.T) {
...
 		var r Timestamptz
 		err := r.Set(tt.source)
-		if err != nil {
-			t.Errorf("%d: %v", i, err)
-		}
-
-		if !r.Equal(&tt.result) {
-			t.Errorf("%d: %v != %v", i, r, tt.result)
-		}
+		require.NoError(t, err)
```
Comment (yevgenypats) [inline schema/timestamptz_test.go:42]: Let's not use testify.

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Let's avoid using testify it doesn't bring a lot of value and can use regular `if err` where we can also use better error messages.

File: schema/timestamptz_test.go (hunk 4 above)
Comment (erezrokah) [reply schema/timestamptz_test.go:42]: Any reason not to use it? It prints better error messages (a better diff between the expected and actual values)

Comment (erezrokah) [reply schema/timestamptz_test.go]: I reverted that change as it's unrelated to this PR. However we do use `testify` in many other places

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**